cluegonauts/clueless/tests.py

```diff
@@ -1,4 +1,73 @@
 # Use noqa: F401 to ignore unused import error
 from django.test import TestCase  # noqa: F401
+from channels.testing import WebsocketCommunicator
+from channels.testing import ChannelsLiveServerTestCase
+from selenium.webdriver.common.by import By
+from selenium.webdriver.firefox.webdriver import WebDriver
+from clueless.consumers import GamePlayersConsumer
 
-# Create your tests here.
+
+class ConsumerTests(TestCase):
+    # Tests consumer's initial connection
+    async def test_quick_game_connection(self):
+        client1_communicator = WebsocketCommunicator(GamePlayersConsumer.as_asgi(),  r"ws/clueless/gameroom/$")
+        client1_connected, subprotocol = await client1_communicator.connect()
+        assert client1_connected
+
+        client2_communicator = WebsocketCommunicator(GamePlayersConsumer.as_asgi(), r"ws/clueless/gameroom/$")
+        client2_connected, subprotocol = await client2_communicator.connect()
+        assert client2_connected
+
+        await client1_communicator.send_json_to({'message': 'Miss Scarlett has joined the game.'})
+
+        client1_response = await client1_communicator.receive_json_from()
+        assert client1_response['message'] == 'Miss Scarlett has joined the game.'
+        client2_response = await client2_communicator.receive_json_from()
+        assert client2_response['message'] == 'Miss Scarlett has joined the game.'
+
+        await client1_communicator.disconnect()
+        await client2_communicator.disconnect()
+
+
+class SeleniumTests(ChannelsLiveServerTestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.selenium = WebDriver()
```

Comment:
You can add an option to run the browser in headless mode so it doesn't open a browser window when you're running tests, which might be a slightly nicer experience. See https://www.selenium.dev/documentation/webdriver/browsers/firefox/